R/html-tweak.R

```diff
@@ -124,17 +124,30 @@ tweak_rmarkdown_html <- function(html, input_path) {
   invisible()
 }
 
-tweak_homepage_html <- function(html, strip_header = FALSE) {
+tweak_homepage_html <- function(html, strip_header = FALSE, sidebar = TRUE) {
   badges <- badges_extract(html)
-  if (length(badges) > 0) {
-    list <- sidebar_section("Dev status", badges)
-    list_div <- paste0("<div>", list, "</div>")
-    list_html <- list_div %>% xml2::read_html() %>% xml2::xml_find_first(".//div")
-
-    sidebar <- html %>% xml2::xml_find_first(".//div[@id='pkgdown-sidebar']")
-    list_html %>%
-      xml2::xml_children() %>%
-      purrr::walk(~ xml2::xml_add_child(sidebar, .))
+
+  # Only add badges if there is a sidebar,
+  # if there are badges,
+  # if there is a place-holder dev section
+  if (sidebar) {
```

Comment:
This is another place where an early return strategy is cleaner:

```R
if (!sidebar) {
  return()
}

...
if (inherits(dev_status_html, "xml_missing)) {
  return()
}
```

Then move the `badges` definition closer to where it's used.